app/ts-cli/geminicli/export.go

```diff
@@ -0,0 +1,1786 @@
+	}
+	d.measurement = mstFilter
+	return nil
+}
+
+// timeFilter [startTime, endTime]
+func (d *dataFilter) timeFilter(t int64) bool {
+	return t >= d.startTime && t <= d.endTime
+}
+
+func (d *dataFilter) isBelowMinTimeFilter(t int64) bool {
+	return t < d.startTime
+}
+
+func (d *dataFilter) isAboveMaxTimeFilter(t int64) bool {
+	return t > d.endTime
+}
+
+type DatabaseDiskInfo struct {
+	dbName          string              // ie. "NOAA_water_database"
+	rps             map[string]struct{} // ie. ["0:autogen","1:every_one_day"]
+	dataDir         string              // ie. "/tmp/openGemini/data/data/NOAA_water_database"
+	walDir          string              // ie. "/tmp/openGemini/data/wal/NOAA_water_database"
+	rpToTsspDirMap  map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/data/NOAA_water_database/0/autogen"}
+	rpToWalDirMap   map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/wal/NOAA_water_database/0/autogen"}
+	rpToIndexDirMap map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/data/NOAA_water_database/0/autogen/index"}
+}
+
+func newDatabaseDiskInfo() *DatabaseDiskInfo {
+	return &DatabaseDiskInfo{
+		rps:             make(map[string]struct{}),
+		rpToTsspDirMap:  make(map[string]string),
+		rpToWalDirMap:   make(map[string]string),
+		rpToIndexDirMap: make(map[string]string),
+	}
+}
+
+func (d *DatabaseDiskInfo) init(actualDataDir string, actualWalDir string, databaseName string, retentionPolicy string) error {
+	d.dbName = databaseName
+
+	// check whether the database is in actualDataPath
+	dataDir := path.Join(actualDataDir, databaseName)
+	if _, err := os.Stat(dataDir); err != nil {
+		return err
+	}
+	// check whether the database is in actualWalPath
+	walDir := path.Join(actualWalDir, databaseName)
+	if _, err := os.Stat(walDir); err != nil {
+		return err
+	}
+
+	// ie. /tmp/openGemini/data/data/my_db  /tmp/openGemini/data/wal/my_db
+	d.dataDir, d.walDir = dataDir, walDir
+
+	ptDirs, err := os.ReadDir(d.dataDir)
+	if err != nil {
+		return err
+	}
+	for _, ptDir := range ptDirs {
+		// ie. /tmp/openGemini/data/data/my_db/0
+		ptTsspPath := path.Join(d.dataDir, ptDir.Name())
+		// ie. /tmp/openGemini/data/wal/my_db/0
+		ptWalPath := path.Join(d.walDir, ptDir.Name())
+
+		if retentionPolicy != "" {
+			ptWithRp := ptDir.Name() + ":" + retentionPolicy
+			rpTsspPath := path.Join(ptTsspPath, retentionPolicy)
+			if _, err := os.Stat(rpTsspPath); err != nil {
+				return fmt.Errorf("retention policy %q invalid : %s", retentionPolicy, err)
+			} else {
+				d.rps[ptWithRp] = struct{}{}
+				d.rpToTsspDirMap[ptWithRp] = rpTsspPath
+				d.rpToIndexDirMap[ptWithRp] = path.Join(rpTsspPath, "index")
+			}
+			rpWalPath := path.Join(ptWalPath, retentionPolicy)
+			if _, err := os.Stat(rpWalPath); err != nil {
+				return fmt.Errorf("retention policy %q invalid : %s", retentionPolicy, err)
+			} else {
+				d.rpToWalDirMap[ptWithRp] = rpWalPath
```

Comment:
If `os.Stat` result is nil, `d.rpToWalDirMap` will not set a value. Is this expected?